### PR TITLE
mockgcp: ComputeManagedSSLCertificate

### PR DESCRIPTION
**Pull request by acpana**
Breaking up from https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/936 for the mockgcp of `ComputeManagedSSLCertificate`. Did a few integration changes but mostly cherry picking Justin's changes 🙏🏼

### CODE REVIEW

File: mockgcp/mockcompute/globalsslcertificatesv1.go
Comment (yuwenma) [inline mockgcp/mockcompute/globalsslcertificatesv1.go:1]: qq: I don't see update method, nor update.yaml test file. Is that intended?

File: mockgcp/mockcompute/globalsslcertificatesv1.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	obj.SelfLink = PtrTo("https://compute.googleapis.com/compute/v1/" + name.String())
+	obj.CreationTimestamp = PtrTo(s.nowString())
+	obj.Id = &id
+	obj.Kind = PtrTo("compute#sslCertificate")
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating sslCertificate: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
```
Comment (yuwenma) [inline mockgcp/mockcompute/globalsslcertificatesv1.go:73]: Suggest using `s.StartLRO`

File: mockgcp/mockcompute/globalsslcertificatesv1.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	deleted := &pb.SslCertificate{}
+	if err := s.storage.Delete(ctx, fqn, deleted); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "sslCertificate %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting sslCertificate: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
```
Comment (yuwenma) [inline mockgcp/mockcompute/globalsslcertificatesv1.go:94]: ditto

File: mockgcp/mockcompute/globalsslcertificatesv1.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			return nil, err
+		}
+
+		name := &globalSSLCertificateName{
+			Project: project,
+			Name:    tokens[4],
+		}
+
+		return name, nil
+	} else {
```
Comment (yuwenma) [inline mockgcp/mockcompute/globalsslcertificatesv1.go:123]: super nit:  no else

File: pkg/test/resourcefixture/testdata/basic/compute/v1alpha1/computemanagedsslcertificate/managedsslcertificate/_http.log [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+Vary: Origin
+Vary: X-Origin
+Vary: Referer
+X-Content-Type-Options: nosniff
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "id": "000000000000000000000",
+  "insertTime": "2024-04-01T12:34:56.123456Z",
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/compute/v1alpha1/computemanagedsslcertificate/managedsslcertificate/_http.log:118]: interesting, do you know the difference between `insertTime` and `createTime`

File: mockgcp/mockcompute/globalsslcertificatesv1.go
Comment (acpana) [reply mockgcp/mockcompute/globalsslcertificatesv1.go:1]: Yes there is no `update`/ `patch` for this resource AFAICT.

File: mockgcp/mockcompute/globalsslcertificatesv1.go (hunk 1 above)
Comment (acpana) [reply mockgcp/mockcompute/globalsslcertificatesv1.go]: This was mostly cherry-picked but thanks for pointing that out 💯 we should probably clarify in the guide when to use `newLRO` vs `startGlobalLRO`

File: pkg/test/resourcefixture/testdata/basic/compute/v1alpha1/computemanagedsslcertificate/managedsslcertificate/_http.log (hunk 4 above)
Comment (acpana) [reply pkg/test/resourcefixture/testdata/basic/compute/v1alpha1/computemanagedsslcertificate/managedsslcertificate/_http.log:122]: Not quite certain, maybe it is resource specific?

**Comment by yuwenma** [comment]
/lgtm
/approve

Thank you! Looks great!

**Pull request merged**